src/core/currency/wallet/currency-wallet-api.js

```diff
@@ -394,15 +395,12 @@ export function makeCurrencyWalletApi(
     },
 
     async getReceiveAddress(
-      opts: EdgeCurrencyCodeOptions = {}
+      opts: EdgeGetReceiveAddressOptions = {}
     ): Promise<EdgeReceiveAddress> {
       const freshAddress = await engine.getFreshAddress(opts)
       const receiveAddress: EdgeReceiveAddress = {
-        metadata: fakeMetadata,
-        nativeAmount: '0',
-        publicAddress: freshAddress.publicAddress,
-        legacyAddress: freshAddress.legacyAddress,
-        segwitAddress: freshAddress.segwitAddress
+        ...freshAddress,
+        metadata: fakeMetadata
```

Comment:
If we want the `nativeAmount` to be mandatory, existing plugins won't be sending that over, so we are going to have a problem. You can do a run-time fallback of `{ nativeAmount: '0', ...freshAddress, metadata: fakeMetadata }` to correct this.

However, as mentioned in the other comment, I think we should do `nativeBalance?: string` to bypass the whole problem.

Reply:
I agree. Moved to nativeBalance